src/externalDevice/BleChessProtocol.ts

```diff
@@ -0,0 +1,257 @@
+import { BaseProtocol, BaseState } from './BaseProtocol'
+import { isUciWithPromotion, isCentralStateCreated, genFullFen, lastMoveToUci, getCommandParams, sendCommandToPeripheral, sendMoveToCentral, areFensSame } from './utils'
+import { State, makeDefaults } from '../chessground/state'
+import { Toast } from '@capacitor/toast'
+import i18n from '../i18n'
+
+export class BleChessProtocol extends BaseProtocol {
+  roundState = makeDefaults()
+  features = new BleChessFeatures
+
+  init(st: State) {
+    this.roundState = st
+    this.transitionTo(new Init)
+  }
+}
+
+class BleChessFeatures {
+  msg: boolean = false
+  lastMove: boolean = false
+}
+
+abstract class BleChessState extends BaseState {
+  setState(state: State) {
+    this.context.roundState = state
+  }
+  getState(): State {
+    return this.context.roundState
+  }
+  getFeatures(): BleChessFeatures {
+    return this.context.features
+  }
+
+  onPeripheralCommand(cmd: string) {
+    sendCommandToPeripheral('nok')
+    Toast.show({ text: `${i18n('unexpected')}: ${cmd}` })
+  }
+  onCentralStateCreated(st: State) {
+    this.setState(st)
+  }
+}
+
+class Init extends BleChessState {
+  onEnter() {
+    this.transitionTo(new CheckFeatureMsg)
+  }
+}
+
+class CheckFeatureMsg extends BleChessState {
+  onEnter() {
+    sendCommandToPeripheral('feature msg')
+  }
+  onPeripheralCommand(cmd: string) {
+    if (cmd === 'ok') {
+      this.getFeatures().msg = true
+      this.transitionTo(new CheckFeatureLastMove)
+    }
+    else if (cmd === 'nok') {
+      this.transitionTo(new CheckFeatureLastMove)
+    }
+    else super.onPeripheralCommand(cmd)
+  }
+}
+
+class CheckFeatureLastMove extends BleChessState {
+  onEnter() {
+    sendCommandToPeripheral('feature last_move')
+  }
+  onPeripheralCommand(cmd: string) {
+    const isCreated = isCentralStateCreated(this.getState())
```

Comment:
can you explain isCreated?

Reply:
it means that game whas started bifore we connect to bluetooth peripheral

Reply:
Maybe let's rename it to isGameStartedBefore

Reply:
`isCentralStateCreated()` has better naming consistency with  `onCentralStateCreated()`  and `onCentralStateChanged()`